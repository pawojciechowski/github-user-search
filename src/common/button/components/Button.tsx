import styled from 'styled-components';
import { Theme } from 'modules/themes/types';
import { baseFontCSS } from 'modules/themes/styles';

const Button = styled.button`
  ${baseFontCSS}
  font-size: 12px;
  line-height: 12px;
  padding: ${({ theme }: { theme: Theme }) => `${theme.spacer * 0.75}px`};
  background: ${({ theme }: { theme: Theme }) => theme.button.background};
  color: ${({ theme }: { theme: Theme }) => theme.button.text.color};
  border-radius: 8px;
  font-weight: bold;
  white-space: nowrap;
  border: none;
  cursor: pointer;

  transition: color 0.15s ease-in-out, background 0.15s ease-in-out;

  &[disabled] {
    background: ${({ theme }: { theme: Theme }) => theme.button.disabled?.background};
    color: ${({ theme }: { theme: Theme }) => theme.button.disabled?.text.color};
  }
`;

export default Button;
