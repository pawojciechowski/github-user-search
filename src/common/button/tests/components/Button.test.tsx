import React from 'react';
import { render } from '@testing-library/react';
import Button from '../../components/Button';
import { testTheme } from 'config/tests/utils';

describe('<Button />', () => {
  it('matches snapshot', () => {
    const { container } = render(<Button theme={testTheme} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
