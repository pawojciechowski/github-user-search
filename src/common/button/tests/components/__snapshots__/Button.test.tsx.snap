// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
.c0 {
  font-family: test;
  font-size: test;
  line-height: test;
  font-weight: test;
  -webkit-letter-spacing: test;
  -moz-letter-spacing: test;
  -ms-letter-spacing: test;
  letter-spacing: test;
  color: test;
  font-size: 12px;
  line-height: 12px;
  padding: 0.75px;
  background: test;
  color: test;
  border-radius: 8px;
  font-weight: bold;
  white-space: nowrap;
  border: none;
  cursor: pointer;
  -webkit-transition: color 0.15s ease-in-out,background 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background 0.15s ease-in-out;
}

.c0[disabled] {
  background: testDisabled;
  color: testDisabled;
}

<button
  class="c0"
/>
`;
